File: backend/DTO/Car/UpdateCarDto.cs
using System.ComponentModel.DataAnnotations;
using DreamBid.Validation;

namespace DreamBid.Dtos.Car
{
    public class UpdateCarDto
    {
        [MaxLength(50)]
        public string? Make { get; set; } = null;

        [MaxLength(50)]
        public string? Model { get; set; } = null;

        [CarManufacturedYear(true)]
        public int? Year { get; set; } = null;

        [Range(0, 1000000)]
        public double? Mileage { get; set; } = null;

        [MaxLength(20)]
        public string? VIN { get; set; } = null;


        public string? ConditionReport { get; set; } = null;

        public double? StartingPrice { get; set; } = null;

        public double? ReservePrice { get; set; } = null;
    }
}

File: frontend/src/dto/car/updateCarDto.js
import Joi from "joi";

class UpdateCarDto{

    make = "";
    model = "";
    year = "";
    mileage ="";
    vin = "";
    conditionReport = "";
    startingPrice = "";
    reservePrice = "";



}

export function validateUpdateCarDto(updateCarDto){

    const schema = Joi.object({
   
        make: Joi.string().max(50),
        model: Joi.string().max(50),
        year: Joi.number().integer().min(1886).max(new Date().getFullYear()),
        mileage: Joi.number().min(0).max(1000000),
        vin: Joi.string().max(20),
        conditionReport: Joi.string(),
        startingPrice: Joi.number(),
        reservePrice: Joi.number(),
    })
    return schema.validate(updateCarDto);

}
export default UpdateCarDto;